lib/include/hse_ikvdb/sched_sts.h

```diff
@@ -1,69 +1,81 @@
 /* SPDX-License-Identifier: Apache-2.0 */
 /*
- * Copyright (C) 2015-2021 Micron Technology, Inc.  All rights reserved.
+ * Copyright (C) 2015-2022 Micron Technology, Inc.  All rights reserved.
  */
 
 #ifndef HSE_IKVDB_STS_H
 #define HSE_IKVDB_STS_H
 
 #include <hse_util/platform.h>
 #include <hse_util/workqueue.h>
+#include <hse_util/list.h>
 
 /* MTF_MOCK_DECL(sched_sts) */
 
 struct sts;
 struct sts_job;
-struct kvdb_rparams;
 
 typedef void sts_job_fn(struct sts_job *job);
-typedef void sts_cancel_fn(struct sts_job *job);
+typedef int sts_print_fn(struct sts_job *job, char *buf, size_t bufsz, const char *fmt);
 
 /**
  * struct sts_job - short term scheduler job handle
- * @sj_link:      for keeping jobs on a linked list (scheduler internal use)
- * @sj_sts:       handle to scheduler that manages this job
+ * @sj_runq_link: run queue list linkage
+ * @sj_print_fn:  callack to print job status line
  * @sj_job_fn:    job handler, set by client, invoked by scheduler
- * @sj_cancel_fn: job cancel function, set by client, invoked by scheduler
- * @sj_tag:       tag, set by client, used to cancel all jobs with same tag
- * @sj_qnum:      scheduler queue
+ * @sj_id:        job ID
+ * @sj_status:    job status code
+ *
+ * A job is initialized with status code 'I' (idle) and then set to 'R'
```

Comment:
I don't really like the usage of `char` here. Could you create an enum instead? At least with `enum`, you get a tiny (using the term liberally) bit more type-safety

Reply:
It's designed so that we can set it to any printable character, but if we use an enum then it adds a bunch of overhead that doesn't seems to buy us anything, yet makes dropping a new status-set call in the code more complicated than necessary.  There are no type-safety issues, it's just an indicator for the sts rest command to display.  Can you make a better case for changing to enum?

Reply:
When I see `char status` in the arguments of the function, I have no idea what valid values of `status` are. In order to find out, I have to read through the implementation of `sched_sts` or alternatively hope that they are well documented somewhere. If I see `enum sched_sts_status`, then I know immediately what the valid statuses are. I can use my editor/IDE to quickly navigate between the various symbols and the `enum`.

If you feel like an enum is heavy-handed, typedefing char and then #defining valid values would be _almost_ equally useful.

Reply:
In my later commit to sched_sts.h I have a comment that describes the status char that I think is pretty clear, no?

Reply:
Lemme add "to any alphabetical character" to that comment...

Reply:
Need to fix ` , ` in the comment. I'll let others comment on it. At the very least, there should be an invariant to check that the status is one of X various characters.

Reply:
Those of us without IDEs don't notice such inconveniences because everything's manual :)

Reply:
You emacs people need to use some language servers and tree-sitter :)